Detectors/ITSMFT/ITS/macros/EVE/DisplayEvents.C

```diff
@@ -43,65 +48,226 @@ class Data
+  }
+  void setDigiTree(TTree* t) { mDigiTree = t; }
+  void setClusTree(TTree* t);
+  void setTracTree(TTree* t);
 
  private:
   // Data loading members
-  static TTree* gDigiTree;
-  static TTree* gClusTree;
-  static TTree* gTracTree;
+  int mLastEvent = 0;
+  PixelReader* mPixelReader = nullptr;
+  ChipPixelData mChipData;
+  o2::InteractionRecord mIR;
   std::vector<Digit> mDigits;
-  std::vector<Cluster> mClusters;
-  std::vector<o2::ITS::TrackITS> mTracks;
-  template <typename T>
-  static void load(TTree* tree, const char* name, int entry, std::vector<T>* arr);
+  std::vector<Cluster>* mClusterBuffer = nullptr;
+  gsl::span<Cluster> mClusters;
+  std::vector<o2::ITSMFT::ROFRecord> mClustersROF;
+  std::vector<o2::ITS::TrackITS>* mTrackBuffer = nullptr;
+  gsl::span<o2::ITS::TrackITS> mTracks;
+  std::vector<o2::ITSMFT::ROFRecord> mTracksROF;
+  void loadDigits();
+  void loadDigits(int entry);
+  void loadClusters(int entry);
+  void loadTracks(int entry);
+
+  TTree* mDigiTree = nullptr;
+  TTree* mClusTree = nullptr;
+  TTree* mTracTree = nullptr;
 
   // TEve-related members
-  static TEveElementList* gEvent;
-  static TEveElementList* gChip;
-  static TEveElement* getEveChipDigits(int chip, const std::vector<Digit>& digi);
-  static TEveElement* getEveChipClusters(int chip, const std::vector<Cluster>& clus);
-  static TEveElement* getEveClusters(const std::vector<Cluster>& clus);
-  static TEveElement* getEveTracks(const std::vector<Cluster>& clus, const std::vector<o2::ITS::TrackITS>& trks);
-};
-
-TTree* Data::gDigiTree = nullptr;
-TTree* Data::gClusTree = nullptr;
-TTree* Data::gTracTree = nullptr;
-TEveElementList* Data::gEvent = nullptr;
-TEveElementList* Data::gChip = nullptr;
-
-template <typename T>
-void Data::load(TTree* tree, const char* name, int entry, std::vector<T>* arr)
+  TEveElementList* mEvent = nullptr;
+  TEveElementList* mChip = nullptr;
+  TEveElement* getEveChipDigits(int chip);
+  TEveElement* getEveChipClusters(int chip);
+  TEveElement* getEveClusters();
+  TEveElement* getEveTracks();
+} evdata;
+
+void Data::loadDigits()
+{
+  auto ir = mChipData.getInteractionRecord();
+  std::cout << "orbit/crossing: " << ' ' << ir.orbit << '/' << ir.bc << '\n';
+
+  mDigits.clear();
+
+  do {
+    auto chipID = mChipData.getChipID();
+    auto pixels = mChipData.getData();
+    for (auto& pixel : pixels) {
+      auto col = pixel.getCol();
+      auto row = pixel.getRow();
+      mDigits.emplace_back(chipID, 0, row, col);
+    }
+    mChipData.clear();
+    if (!mPixelReader->getNextChipData(mChipData))
+      return;
+    ir = mChipData.getInteractionRecord();
+  } while (mIR == ir);
+  mIR = ir;
+
+  std::cout << "Number of ITSDigits: " << mDigits.size() << '\n';
+}
+
+void Data::loadDigits(int entry)
+{
+  if (mPixelReader == nullptr)
+    return;
+
+  for (; mLastEvent < entry; mLastEvent++) {
+    auto ir = mChipData.getInteractionRecord();
+    do {
+      mChipData.clear();
```

Comment:
idem

Reply:
@shahor02 Indeed, this was not really needed. Thank you. Fixed.